diff --git a/device-common.mk b/device-common.mk
index d3ecf60..9b12c64 100644
--- a/device-common.mk
+++ b/device-common.mk
@@ -65,6 +65,13 @@ PRODUCT_COPY_FILES +=  \
         frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml \
         frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml
 
+#Copy Graphics binaries
+PRODUCT_COPY_FILES += $(call add-to-product-copy-files-if-exists,\
+                       vendor/mali/64bit/libGLES_mali.so:system/lib64/egl/libGLES_mali.so\
+                       vendor/mali/64bit/gralloc.hikey.so:system/lib64/hw/gralloc.hikey.so\
+                       vendor/mali/32bit/libGLES_mali.so:system/lib/egl/libGLES_mali.so \
+                       vendor/mali/32bit/gralloc.hikey.so:system/lib/hw/gralloc.hikey.so)
+
 # Include BT modules
 $(call inherit-product-if-exists, device/linaro/hikey/wpan/ti-wpan-products.mk)
 
